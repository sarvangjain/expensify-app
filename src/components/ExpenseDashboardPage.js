import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilters';
import ExpenseSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
  <div>
    This is from my dashboard component!
    <ExpenseSummary />
    <ExpenseListFilter />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
