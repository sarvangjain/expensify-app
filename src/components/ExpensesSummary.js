import {connect} from 'react-redux';
import SelectExpensesTotal from '../selectors/expenses-total';
import SelectExpenses from '../selectors/expenses';
import React from 'react';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
    <div>
        {
            props.count > 0 ? 
            <h2>
                Viewing {props.count} 
                {props.count === 1 ?  ' expense ' : ' expenses '} 
                with total expenditure of {numeral(props.total/100).format('$0,0.00')}
            </h2>
            : 
                <h2>
                Viewing 0 expenses
            </h2>
        }
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = SelectExpenses(state.expenses, state.filters);
    return {
        total: SelectExpensesTotal(visibleExpenses),
        count: visibleExpenses.length
    };
};

export default connect(mapStateToProps)(ExpenseSummary);