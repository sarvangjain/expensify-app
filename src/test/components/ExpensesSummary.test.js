import {ExpenseSummary} from '../../components/ExpensesSummary';
import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

test('should handle ExpenseSummary for 0 expenses', () => {
    const wrapper = shallow(<ExpenseSummary count={0} total={0}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should handle ExpenseSummary for 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary total={195} count={1}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should handle ExpenseSummary for multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary total={5000} count={6}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});