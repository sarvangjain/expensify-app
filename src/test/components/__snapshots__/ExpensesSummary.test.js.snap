// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle ExpenseSummary for 0 expenses 1`] = `
<div>
  <h2>
    Viewing 0 expenses
  </h2>
</div>
`;

exports[`should handle ExpenseSummary for 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     expense 
    with total expenditure of 
    $1.95
  </h2>
</div>
`;

exports[`should handle ExpenseSummary for multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    6
     expenses 
    with total expenditure of 
    $50.00
  </h2>
</div>
`;
